physics/cires_ugwp_v1.F90

```diff
@@ -0,0 +1,635 @@
+!! Unified Formalism:
+!! 1. GW Sources: Stochastic and physics based mechanisms for GW-excitations in the lower atmosphere, calibrated by the high-res analyses/forecasts, and observations (3 types of GW sources: orography, convection, fronts/jets).
+!! 2. GW Propagation: Unified solver for "propagation, dissipation and breaking" excited from all type of GW sources.
+!! 3. GW Effects: Unified representation of GW impacts on the "resolved" flow for all sources (energy-balanced schemes for momentum, heat and mixing).
+!!
+!! https://www.weather.gov/media/sti/nggps/Presentations%202017/02%20NGGPS_VYUDIN_2017_.pdf
+
+module cires_ugwp_v1
+
+    use machine, only: kind_phys
+
+    use cires_ugwp_module_v1, only: knob_ugwp_version, cires_ugwp_init_v1, &
+                                 cires_ugwp_finalize_v1
+
+    use gwdps, only: gwdps_run
+
+    implicit none
+
+    private
+
+    public cires_ugwp_v1_run
+
+    logical :: is_initialized = .False.
+
+
+contains
+
+
+
+! -----------------------------------------------------------------------
+! CCPP entry points for CIRES Unified Gravity Wave Physics (UGWP) scheme v0
+! -----------------------------------------------------------------------
+!    originally from cires_ugwp_driver_v1.f
+!    driver of cires_ugwp   (_driver)
+! -----------------------------------------------------------------------
+!   driver is called after pbl & before chem-parameterizations
+! -----------------------------------------------------------------------
+!  order = dry-adj=>conv=mp-aero=>radiation -sfc/land- chem -> vertdiff-> [rf-gws]=> ion-re
+! -----------------------------------------------------------------------
+!>@brief The subroutine initializes the CIRES UGWP
+!> \section arg_table_cires_ugwp_init Argument Table
+!! \htmlinclude cires_ugwp_v1_init.html
+!!
+! -----------------------------------------------------------------------
+!
+
+
+subroutine cires_ugwp_v1_init(Model,Init_parm,Grid)
+                ! NOTE:  This code is taken from Valery's code in subroutine
+                ! GFS_initialize in GFS_driver.F90
+
+use cires_ugwp_module,   only: cires_ugwp_init_emc, cires_ugwp_init_v1, cires_indx_ugwp
+
+implicit none
+
+real(kind=kind_phys), parameter   :: p_ref = 101325.0d0
+
+
+!----  initialization of cires_ugwp .
+!     if ( Model%me == Model%master) print *,  ' VAY-nml ',  Model%fn_nml
+!     if ( Model%me == Model%master) print *,  ' VAY-nml2 ', Model%input_nml_file
+    if (Model%cdmbgwd(3) > 0.0) then 
+!     if ( Model%me == Model%master) print *,  ' VAY-nml ',  Model%fn_nml,
+!     Model%input_nml_file
+     call cires_ugwp_init_emc(Model%me,      Model%master, Model%nlunit,  Init_parm%logunit, &
+                           Model%fn_nml,  Model%lonr,   Model%latr,    Model%levs,          &    
+                           Init_parm%ak,  Init_parm%bk, p_ref,         Model%dtp,           &    
```

Comment:
Use the Model%ak and Model%bk version, not Init_parm THe standard names in GFS_typedefs.meta should already point to Model%ak and Model%bk (I hope so at least!)